pkg/api/types.go

```diff
@@ -518,8 +518,18 @@ type GitRepoVolumeSource struct {
 // The contents of the target Secret's Data field will be presented in a volume
 // as files using the keys in the Data field as the file names.
 type SecretVolumeSource struct {
-	// Name of the secret in the pod's namespace to use
-	SecretName string `json:"secretName"`
+	// Name of the secret in the pod's namespace to use.  May not be specified if items is specified.
+	SecretName string `json:"secretName,omitempty"`
+	// A list of projections to make into the volume.  May not be specified if SecretName is specified.
+	Items []SecretVolumeFile `json:"items,omitempty"`
+}
+
+// SecretVolumeFile represents a single secret key projected into a file.
+type SecretVolumeFile struct {
+	// A reference to the secret key to project into the volume.
+	SecretKeySelector `json:",inline"`
```

Comment:
Two observations on this for discussion.

1) This allows multiple Secrets to project into a single mount.  I don't have a problem with this.

2) This makes the case of projecting multiple keys from a single Secret harder in that you have to name the Secret over and over.  I feel like this would be a common case, especially for ConfigMap which should probably follow the same pattern as Secret.

How bad would it be for the volume to look like:

```
   name: MySecrets
   secret:
      secretName: app-secret
      items:
         - key: username
           path: username.txt
         - key: password
           path: password.txt
```

(note this breaks the multi-secret-in-one-mount property)

and the env to look like:

```
  - name: USERNAME
    valueFrom:
      secretRef:
        secretName: app-secret
        key: username
```

(note that this means they use different structs)